### PR TITLE
[driver] Do not warn about unused plugin flags.

### LINKED ISSUE #88173: Clang produces error on compiling assembly files when using plugins like polly
Suppose we have a simple test program called test.cxx and we have polly plugin LLVMPolly.so built dynamically along with LLVM (with `LLVM_POLLY_LINK_INTO_TOOLS=OFF`):

```
#include <iostream>

int main() {
	std::cout << "Hello world!" << std::endl;
        return 0;
}
```
Compiling it with `clang++ -fplugin=LLVMPolly.so -mllvm -polly -mllvm -polly-parallel -o test test.cxx` works as expected, but what happens when compiling assembly?

To test it, we create our test assembly `test.S` with `clang++ -fplugin=LLVMPolly.so -mllvm -polly -mllvm -polly-parallel -S -c -o test.S test.cxx` and then we try to compile `test.S` as object, then we get:

```
clang++ -fplugin=LLVMPolly.so -mllvm -polly -mllvm -polly-parallel -c -o test.S.o test.S 
clang (LLVM option parsing): Unknown command line argument '-polly'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--color'?
clang (LLVM option parsing): Unknown command line argument '-polly-parallel'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--ppc-gen-isel'?
```

It seems that it is not loading the plugin and not recognizing the options that worked in previous phases (I assume because they are not needed/used when compiling assembly)

I think this is a bug because in this case it is much better to load and ignore plugin options if they are not used rather than throw an error because polly use flags might be present in `C{XX}FLAGS` making the whole target program to fail to compile if it includes assembly files.

One real example of it is the LLVM library itself which fails to compile when CXXFLAGS with polly options are used on some assembly files that LLVM compiles:

```
[126/3501 (  3%)] /usr/lib/ccache/bin/clang -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm_build-abi_x86_64.amd64/lib/Support/BLAKE3 -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm/lib/Support/BLAKE3 -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm_build-abi_x86_64.amd64/include -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm/include  -DNDEBUG -O2 -pipe -march=native -Wno-unused-command-line-argument -O3 -fplugin=LLVMPolly.so -mllvm -polly -mllvm -polly-parallel -mllvm -polly-vectorizer=stripmine -mllvm -polly-run-dce -mllvm -polly-scheduling-chunksize=1 -mllvm -polly-scheduling=dynamic -mllvm -polly-num-threads=32 -mllvm -polly-omp-backend=LLVM -mllvm -polly-invariant-load-hoisting -fopenmp  -flto=thin -fPIC -MD -MT lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o -MF lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o.d -o lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o -c /tmp/portage/sys-devel/llvm-18.1.3/work/llvm/lib/Support/BLAKE3/blake3_sse2_x86-64_unix.S
FAILED: lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o 
/usr/lib/ccache/bin/clang -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm_build-abi_x86_64.amd64/lib/Support/BLAKE3 -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm/lib/Support/BLAKE3 -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm_build-abi_x86_64.amd64/include -I/tmp/portage/sys-devel/llvm-18.1.3/work/llvm/include  -DNDEBUG -O2 -pipe -march=native -Wno-unused-command-line-argument -O3 -fplugin=LLVMPolly.so -mllvm -polly -mllvm -polly-parallel -mllvm -polly-vectorizer=stripmine -mllvm -polly-run-dce -mllvm -polly-scheduling-chunksize=1 -mllvm -polly-scheduling=dynamic -mllvm -polly-num-threads=32 -mllvm -polly-omp-backend=LLVM -mllvm -polly-invariant-load-hoisting -fopenmp  -flto=thin -fPIC -MD -MT lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o -MF lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o.d -o lib/Support/BLAKE3/CMakeFiles/LLVMSupportBlake3.dir/blake3_sse2_x86-64_unix.S.o -c /tmp/portage/sys-devel/llvm-18.1.3/work/llvm/lib/Support/BLAKE3/blake3_sse2_x86-64_unix.S
clang (LLVM option parsing): Unknown command line argument '-polly'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--color'?
clang (LLVM option parsing): Unknown command line argument '-polly-parallel'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--ppc-gen-isel'?
clang (LLVM option parsing): Unknown command line argument '-polly-vectorizer=stripmine'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--slp-vectorize-hor=stripmine'?
clang (LLVM option parsing): Unknown command line argument '-polly-run-dce'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--unroll-runtime'?
clang (LLVM option parsing): Unknown command line argument '-polly-scheduling-chunksize=1'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--cache-line-size=1'?
clang (LLVM option parsing): Unknown command line argument '-polly-scheduling=dynamic'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--post-RA-scheduler=dynamic'?
clang (LLVM option parsing): Unknown command line argument '-polly-num-threads=32'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--xcore-max-threads=32'?
clang (LLVM option parsing): Unknown command line argument '-polly-omp-backend=LLVM'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--pre-RA-sched=LLVM'?
clang (LLVM option parsing): Unknown command line argument '-polly-invariant-load-hoisting'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--licm-control-flow-hoisting'?
```

For more info look the [Gentoo PR](https://github.com/gentoo/gentoo/pull/36158) I did for trying to package `Polly` for Gentoo

Edit: It is worth to say that this affects all clang versions including 19 git version.

### PR DESCRIPTION
**Pull request by Meinersbur**
Plugins are not loaded without the -cc1 phase. Do not report them when running on an assembly file or when linking. Many build tools add these options to all driver invocations, including LLVM's build system.

Fixes #88173

### CODE REVIEW

File: clang/test/Driver/clang-s-plugin.s [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// RUN: %clang -### -c -fpass-plugin=bar.so -fplugin=bar.so -fplugin-arg-bar-option -Werror %s 2>&1 | FileCheck %s
```
Comment (MaskRay) [inline clang/test/Driver/clang-s-plugin.s:1]: This can be moved to `Driver/plugin-driver-args.cpp` with `%clang -### -c -x assembler`

**Review by MaskRay — APPROVED** [review]

File: clang/test/Driver/clang-s-plugin.s (hunk 1 above)
Comment (Meinersbur) [reply clang/test/Driver/clang-s-plugin.s]: Done

**Comment by Meinersbur** [comment]
I accidentally pushed a Foxtrot merge. Will push once the pre-merge bot is happy.

Thanks for the review.

**Comment by jakeegan** [comment]
Hi, the test plugin-driver-args.cpp is failing on AIX now, could you take a look please?
https://lab.llvm.org/buildbot/#/builders/214/builds/12118/steps/6/logs/FAIL__Clang__plugin-driver-args_cpp

**Comment by Meinersbur** [comment]
@jakeegan  Sorry, didn't see you comment over an extended weekend.

**Pull request merged**